pkg/analysis/numericbounds/analyzer.go

```diff
@@ -0,0 +1,212 @@
+*/
+package numericbounds
+
+import (
+	"errors"
+	"fmt"
+	"go/ast"
+
+	"golang.org/x/exp/constraints"
+	"golang.org/x/tools/go/analysis"
+	kalerrors "sigs.k8s.io/kube-api-linter/pkg/analysis/errors"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/helpers/extractjsontags"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/helpers/inspector"
+	markershelper "sigs.k8s.io/kube-api-linter/pkg/analysis/helpers/markers"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/utils"
+	"sigs.k8s.io/kube-api-linter/pkg/markers"
+)
+
+const (
+	name = "numericbounds"
+)
+
+// Type bounds for validation.
+const (
+	maxInt32   = 2147483647                                      // 2^31 - 1
+	minInt32   = -2147483648                                     // -2^31
+	maxFloat32 = 3.40282346638528859811704183484516925440e+38    // max float32
+	minFloat32 = -3.40282346638528859811704183484516925440e+38   // min float32
+	maxFloat64 = 1.797693134862315708145274237317043567981e+308  // max float64
+	minFloat64 = -1.797693134862315708145274237317043567981e+308 // min float64
+)
+
+// JavaScript safe integer bounds for int64 (±2^53-1).
+// Per Kubernetes API conventions, int64 fields should use bounds within this range
+// to ensure compatibility with JavaScript clients.
+const (
+	maxSafeInt64 = 9007199254740991  // 2^53 - 1 (max safe integer in JS)
+	minSafeInt64 = -9007199254740991 // -(2^53 - 1) (min safe integer in JS)
+)
+
+var errMarkerMissingValue = errors.New("marker value not found")
+
+// Analyzer is the analyzer for the numericbounds package.
+// It checks that numeric fields have appropriate bounds validation markers.
+var Analyzer = &analysis.Analyzer{
+	Name:     name,
+	Doc:      "Checks that numeric fields (int32, int64, float32, float64) have appropriate minimum and maximum bounds validation markers",
+	Run:      run,
+	Requires: []*analysis.Analyzer{inspector.Analyzer},
+}
+
+func run(pass *analysis.Pass) (any, error) {
+	inspect, ok := pass.ResultOf[inspector.Analyzer].(inspector.Inspector)
+	if !ok {
+		return nil, kalerrors.ErrCouldNotGetInspector
+	}
+
+	inspect.InspectFields(func(field *ast.Field, _ extractjsontags.FieldTagInfo, markersAccess markershelper.Markers, qualifiedFieldName string) {
+		// Create TypeChecker with closure capturing markersAccess and qualifiedFieldName
+		// Ignore TypeChecker's prefix since we use qualifiedFieldName from inspector
+		typeChecker := utils.NewTypeChecker(func(pass *analysis.Pass, ident *ast.Ident, node ast.Node, _ string) {
+			checkNumericType(pass, ident, node, markersAccess, qualifiedFieldName)
+		})
+
+		typeChecker.CheckNode(pass, field)
+	})
+
+	return nil, nil //nolint:nilnil
+}
+
+//nolint:cyclop
+func checkNumericType(pass *analysis.Pass, ident *ast.Ident, node ast.Node, markersAccess markershelper.Markers, qualifiedFieldName string) {
+	// Only check int32, int64, float32, and float64 types
+	if ident.Name != "int32" && ident.Name != "int64" && ident.Name != "float32" && ident.Name != "float64" {
+		return
+	}
```

Comment:
This would be neater represented as a switch
```suggestion
	switch ident.Name {
	case "int32", "int64", "float32", "float64":
		// These are the types we want to check.
	default:
		return
	}

```